dbms/src/Storages/Transaction/tests/kvstore_helper.h

```diff
@@ -0,0 +1,154 @@
+#include <Storages/registerStorages.h>
+#include <TestUtils/TiFlashTestBasic.h>
+#include <TestUtils/TiFlashTestEnv.h>
+
+#include <memory>
+
+namespace DB
+{
+extern void GenMockSSTData(const TiDB::TableInfo & table_info,
+                           TableID table_id,
+                           const String & store_key,
+                           UInt64 start_handle,
+                           UInt64 end_handle,
+                           UInt64 num_fields = 1,
+                           const std::unordered_set<ColumnFamilyType> & cfs = {ColumnFamilyType::Write, ColumnFamilyType::Default});
+
+namespace FailPoints
+{
+extern const char skip_check_segment_update[];
+}
+
+namespace RegionBench
+{
+extern void setupPutRequest(raft_cmdpb::Request *, const std::string &, const TiKVKey &, const TiKVValue &);
+extern void setupDelRequest(raft_cmdpb::Request *, const std::string &, const TiKVKey &);
+} // namespace RegionBench
+
+extern void RemoveRegionCommitCache(const RegionPtr & region, const RegionDataReadInfoList & data_list_read, bool lock_region = true);
+extern void CheckRegionForMergeCmd(const raft_cmdpb::AdminResponse & response, const RegionState & region_state);
+extern void ChangeRegionStateRange(RegionState & region_state, bool source_at_left, const RegionState & source_region_state);
+
+namespace tests
+{
+// TODO: Use another way to workaround calling the private methods on KVStore
+class RegionKVStoreTest : public ::testing::Test
+{
+public:
+    RegionKVStoreTest()
+    {
+        test_path = TiFlashTestEnv::getTemporaryPath("/region_kvs_test");
+    }
+
+    static void SetUpTestCase() {}
+
+    void SetUp() override
+    {
+        // clean data and create path pool instance
+        path_pool = createCleanPathPool(test_path);
+
+        reloadKVSFromDisk();
+
+        proxy_instance = std::make_unique<MockRaftStoreProxy>();
+        proxy_helper = std::make_unique<TiFlashRaftProxyHelper>(MockRaftStoreProxy::SetRaftStoreProxyFFIHelper(
+            RaftStoreProxyPtr{proxy_instance.get()}));
+        kvstore->restore(*path_pool, proxy_helper.get());
+        {
+            auto store = metapb::Store{};
+            store.set_id(1234);
+            kvstore->setStore(store);
+            ASSERT_EQ(kvstore->getStoreID(), store.id());
+        }
+    }
+
+    void TearDown() override {}
+
+protected:
+    KVStore & getKVS() { return *kvstore; }
+    KVStore & reloadKVSFromDisk()
+    {
+        kvstore.reset();
+        auto & global_ctx = TiFlashTestEnv::getGlobalContext();
+        kvstore = std::make_unique<KVStore>(global_ctx, TiDB::SnapshotApplyMethod::DTFile_Directory);
+        // only recreate kvstore and restore data from disk, don't recreate proxy instance
+        kvstore->restore(*path_pool, proxy_helper.get());
+        return *kvstore;
+    }
+    void createDefaultRegions()
+    {
+        proxy_instance->init(100);
+    }
+    void initStorages()
+    {
+        registerStorages();
```

Comment:
we need to try ... catch and ignore the exception thrown by this method. Because other unit test may already call `registerStorages` when we run without any gtest-filter